### PR TITLE
Update messages to use PackageAppVersion in response messages

### PR DESCRIPTION
**Pull request by absoludity**
### Description of the change

In [Antonio's comment](https://github.com/kubeapps/kubeapps/issues/3165#issuecomment-882943900) he mentioned that when working on the UX he was missing the latest app version in a couple of places where he needed it.

In some places we have just the package version, in others we're listing two separate fields for the package and app versions. In the response for the `GetAvailablePackageVersions` we use a combined `PackageAppVersion` message to contain both.

The proposal here is to pull out the `PackageAppVersion` message, and use that in all the places where we have a field for the package version so that we always include both the package and app version information without explicitly needing two fields.

Edit: After the conversation on #3283 , I've updated the PR with a mostly mechanical change to use the `PackageAppVersion` message as proposed.

### Benefits

We will have both available wherever we have a field for the package version.

### Applicable issues

  - Ref #3165

### Additional information

If there's agreement, I'll

 - [ ] update the design doc to match and 
 - [x] take care of updating both the helm and flux plugins to this.

### CODE REVIEW

**Comment by absoludity** [comment]
Sorry @gfichtenholt . This is the same as PR #3283 , but GitHub doesn't allow me to create a followup PR targetting kubeapps from my own repo, so I've closed that one and opened this one instead.

File: cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/release_test.go [hunk 1]
```diff
@@ -816,16 +834,20 @@ var redis_summary_latest = &corev1.InstalledPackageSummary{
...
+	},
+	PkgDisplayName:   "redis",
+	ShortDescription: "Open source, advanced key-value store. It is often referred to as a data structure server since keys can contain strings, hashes, lists, sets and sorted sets.",
 	Status: &corev1.InstalledPackageStatus{
 		Ready:      true,
 		Reason:     corev1.InstalledPackageStatus_STATUS_REASON_INSTALLED,
 		UserReason: "ReconciliationSucceeded",
 	},
-	LatestPkgVersion: "14.6.1",
+	LatestVersion: &corev1.PackageAppVersion{
```
Comment (gfichtenholt) [inline cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/release_test.go:848]: yeah, I noticed flux CRD doesn't give me an app version, only pkg version. I guess I will have to pull that from the cached index, if/when needed

File: cmd/kubeapps-apis/proto/kubeappsapis/core/packages/v1alpha1/packages.proto [hunk 2]
```diff
@@ -274,17 +259,11 @@ message AvailablePackageSummary {
...
   // The latest version available for this package. Often expected when viewing
   // a summary of many available packages.
-  string latest_pkg_version = 3;
-
-  // Latest available package app version
-  //
-  // The app version of the latest version available for this package. Often expected
-  // when viewing a summary of many available packages.
-  string latest_app_version = 4;
+  PackageAppVersion latest_version = 3;
```
Comment (gfichtenholt) [inline cmd/kubeapps-apis/proto/kubeappsapis/core/packages/v1alpha1/packages.proto:266]: looks like you skipped number 4. here. probably just a typo

File: cmd/kubeapps-apis/proto/kubeappsapis/core/packages/v1alpha1/packages.proto [hunk 3]
```diff
@@ -328,15 +307,10 @@ message AvailablePackageDetail {
...
-  // The version of the package (e.g. chart version)
-  string pkg_version = 3;
-
-  // Available package app version
+  // Available version
   //
-  // The version of the packaged application (e.g. wordpress version).
-  string app_version = 4;
+  // The version of the package and application.
+  PackageAppVersion version = 3;
```
Comment (gfichtenholt) [inline cmd/kubeapps-apis/proto/kubeappsapis/core/packages/v1alpha1/packages.proto:313]: again, same as above looks like you skipped number 4

**Review by gfichtenholt — APPROVED** [review]
+1 with couple very minor changes. Thanks

File: cmd/kubeapps-apis/proto/kubeappsapis/core/packages/v1alpha1/packages.proto (hunk 2 above)
Comment (absoludity) [reply cmd/kubeapps-apis/proto/kubeappsapis/core/packages/v1alpha1/packages.proto:266]: I just didn't want to bloat the diff (was going to re-order in a separate PR, but happy to do so here). Let me do that.

**Review by gfichtenholt — APPROVED** [review]
LGTM

**Pull request merged**